front/src/WebRtc/VideoPeer.ts

```diff
@@ -69,6 +73,35 @@ export class VideoPeer extends Peer {
             };
         });
 
+        this.volumeStore = readable<number | null>(null, (set) => {
+            let timeout: ReturnType<typeof setTimeout>;
+            const unsubscribe = this.streamStore.subscribe((mediaStream) => {
+                if (mediaStream === null || mediaStream.getAudioTracks().length <= 0) {
+                    set(null);
+                    return;
+                }
+                const soundMeter = new SoundMeter();
+                soundMeter.connectToSource(mediaStream, new AudioContext());
+                let error = false;
+
+                timeout = setInterval(() => {
+                    try {
+                        set(soundMeter.getVolume());
+                    } catch (err) {
+                        if (!error) {
+                            console.error(err);
+                            error = true;
+                        }
+                    }
+                }, 100);
+            });
+
+            return () => {
+                unsubscribe();
+                clearInterval(timeout);
+            };
+        });
```

Comment:
This code is very similar to the one defining the localVolumeStore.

Could we maybe try to make a function that acts as a factory of volume stores and that is used in both places?

Something like:

```
export function createVolumeStore(streamStore: Readable<Stream>): Readable<number | null> {
   // ...
}
```